csharp/ql/src/Concurrency/Concurrency.qll

```diff
@@ -37,22 +40,29 @@ class SynchronizedMethodAttribute extends Attribute {
   }
 }
 
-// A method with attribute [MethodImpl(MethodImplOptions.Synchronized)]
-class SynchronizedMethod extends Method {
+/** A method with attribute `[MethodImpl(MethodImplOptions.Synchronized)]`. */
+private class SynchronizedMethod extends Method {
   SynchronizedMethod() { getAnAttribute() instanceof SynchronizedMethodAttribute }
 
+  /** Holds if the method locks `this`. */
   predicate isLockThis() { not isStatic() }
 
+  /** Gets the type that is locked by this method, if any. */
   Type getLockTypeObject() { isStatic() and result = getDeclaringType() }
 }
 
+/** A block that is locked by a `lock` statement. */
 abstract class LockedBlock extends BlockStmt {
+  /** Holds if the `lock` statement locks `this`. */
   abstract predicate isLockThis();
 
+  /** Gets the lock variable of the `lock` statement, if any. */
   abstract Variable getLockVariable();
 
+  /** Gets the locked type of the `lock` statement, if any. */
```

Comment:
When can a locked expression not have a type? I.e., why the `, if any`?

Reply:
Because it only gives results for `lock(typeof(Foo))`.